src/reducers/index.js

```diff
@@ -90,31 +89,39 @@ export const refreshResourceTemplate = (state, action) => {
  */
 export const setResourceTemplate = (state, action) => {
   const resourceTemplateId = action.payload.id
-  let output = state
+  let newState = { ...state }
+
+  newState.entities.resourceTemplates[resourceTemplateId] = action.payload
 
   action.payload.propertyTemplates.forEach((property) => {
     const propertyAction = {
       payload: {
-        reduxPath: [resourceTemplateId, property.propertyURI],
+        reduxPath: ['resource', resourceTemplateId, property.propertyURI],
         property,
       },
     }
 
-    output = refreshResourceTemplate(output, propertyAction)
+    newState = refreshResourceTemplate(newState, propertyAction)
   })
 
-  output[resourceTemplateId].rdfClass = action.payload.resourceURI
-
-  return output
+  return newState
 }
 
 export const setResourceURI = (state, action) => {
   const newState = { ...state }
-  const reduxPath = action.payload.reduxPath
 
-  const lastObject = reduxPath.reduce((newState, key) => newState[key] = newState[key] || {}, newState)
+  // Call payload just to avoid an eslint warning
+  action.payload
 
-  lastObject.rdfClass = action.payload.resourceURI
+  /*
+   * TODO: fix this
```

Comment:
Just a reminder to fix this or ticket separately.

Reply:
This was the part I was hoping you would fix for me 😆

Reply:
Can't we just get rid of setResourceURI altogether?